src/reactionsystem.jl

```diff
@@ -0,0 +1,127 @@
+using Catalyst, ModelingToolkit, Symbolics
+
+# module SBML
+# struct Model end
+# struct Reaction end
+# struct Compartment end
+# struct Species end
+# end
+
+""" ReactionSystem constructor """
+function ModelingToolkit.ReactionSystem(model::Model)  # Todo: requires unique parameters (i.e. SBML must have been imported with localParameter promotion in libSBML)
+    model = make_extensive(model)
+    model = expand_reversible(model)
+    rxs = mtk_reaction.(model.reactions)
+    t = DEFAULT_IV
+    species = [create_var(k) for k in keys(model.species)]
+    params = vcat([create_var(k) for k in keys(model.parameters)], [create_par(k) for k in keys(model.compartments)])
+    ReactionSystem(rxs,t,species,params)
+end
+
+""" ReactionSystem constructor """
+function ModelingToolkit.ReactionSystem(sbmlfile::String)
+    model = readSBML(sbmlfile)
+    ReactionSystem(model)
+end
+
+""" ODESystem constructor """
+function ModelingToolkit.ODESystem(model::Model)
+    rs = ReactionSystem(model)
+    u0map = get_u0(model)
+    parammap = get_paramap(model)
+    defaults = vcat(u0map, parammap)
+    convert(ODESystem, rs, defaults=defaults)
+end
+
+""" ODESystem constructor """
+function ModelingToolkit.ODESystem(sbmlfile::String)
+    model = readSBML(sbmlfile)
+    ODESystem(model)
+end
+
+""" ODEProblem constructor """
+function ModelingToolkit.ODEProblem(model::Model,tspan)  # Todo: add u0 and parameters argument
+    odesys = ODESystem(model)
+    ODEProblem(odesys, [], tspan)
+end
+
+""" ODEProblem constructor """
+function ModelingToolkit.ODEProblem(sbmlfile::String,tspan)  # Todo: add u0 and parameters argument
+    odesys = ODESystem(sbmlfile)
+    ODEProblem(odesys, [], tspan)
+end
+
+""" Convert intensive to extensive expressions """
+function make_extensive(model)
+    model  # Todo: For spevies with `hOSU=false` divide all occurences in mathematical expressions by compartment size.
+           # Also convert species initialConcentrations to initialAmounts
+end
+
+""" Expand reversible reactions to two reactions """
+function expand_reversible(model)
+    model  # Todo: convert all Reactions that are `reversible=true` to a forward and reverse reaction with `reversible=false`.
```

Comment:
Do we need to do this if `X + Y ↔ XY` is valid catalyst? https://catalyst.sciml.ai/stable/tutorials/basics/#Combining-several-reactions-in-one-line

Alternatively, it could be something that Catalyst needs. It seems like a useful transform to have.

Reply:
Maybe worth pinging Sam about

Reply:
The way as SBML would describe the kineticLaw for a reversible reaction is `X + Y ↔ XY` is `kAs*X*Y-kDi*XY`. Catalyst however needs `kAs*X*Y, kDi*XY`, that is two kineticLaws. I mean this is a simple example, but somehow we need to find a way that generally works in splitting a combined kineticLaw in a forward and reverse kineticLaw. Any suggestions how to do this in a clean way?